tests/ext-popup.test.js

```diff
@@ -0,0 +1,204 @@
+import fs from 'fs';
+import path from 'path';
+import Popup from '../src/lib/ext-popup';
+import * as ExtListen from '../src/lib/ext-listen';
+
+const popupHtml = fs.readFileSync(
+  path.resolve(__dirname, '../src/popup/popup.html'),
+  'utf-8'
+);
+
+const domParser = new DOMParser();
+const popupBody = domParser.parseFromString(popupHtml, 'text/html').body
+  .innerHTML;
+
+function observeChange(element) {
+  return new Promise((resolve) => {
+    const observer = new MutationObserver(() => {
+      observer.disconnect();
+      resolve();
+    });
+    observer.observe(element, { attributes: true });
+  });
+}
+
+test('start monitoring all extensions from the popup', async () => {
+  document.body.innerHTML = popupBody;
+
+  const getMonitorStatusFn = jest.spyOn(ExtListen, 'getMonitorStatus');
+  const startMonitorFn = jest.spyOn(ExtListen, 'startMonitor');
+
+  const popup = new Popup();
+
+  const startBtn = popup.startMonitorAllBtn;
+  const stopBtn = popup.stopMonitorAllBtn;
+  const monitorStatus = popup.monitorStatusText;
+  const errorText = popup.errorMsgText;
+
+  getMonitorStatusFn.mockResolvedValueOnce(false);
+  getMonitorStatusFn.mockResolvedValueOnce(true);
+  startMonitorFn.mockResolvedValue(true);
+
+  await popup.init();
+
+  // popup at initial - no extensions being watched
+  expect(startBtn.getAttribute('disabled')).toBeNull();
+  expect(stopBtn.getAttribute('disabled')).toEqual('disabled');
```

Comment:
Nit, as an alternative this assertion could use [`hasAttribute`](https://developer.mozilla.org/en-US/docs/Web/API/Element/hasAttribute) or assert the value set on the [InputElement's `disabled` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement) (no really needed to changed, I just wanted to mention it)

Reply:
Thanks for telling this. I think `hasAttribute` is more appropriate in this case, so I made these changes with `hasAttribute`